api/backup/import.go

```diff
@@ -0,0 +1,167 @@
+				return err
+			}
+
+			bucket := tx.Bucket([]byte(bucketName))
+
+			switch bucketName {
+			case "version":
+				versions, ok := v.(map[string]interface{})
+				if !ok {
+					logrus.WithField("obj", v).Errorf("failed to cast %s map[string]interface{}", bucketName)
+				} else {
+					// TODO: test if those exist...
+					Put(bucketName, bucket, "DB_VERSION", versions["DB_VERSION"])
+					Put(bucketName, bucket, "INSTANCE_ID", versions["INSTANCE_ID"])
+				}
+			case "dockerhub":
+				Put(bucketName, bucket, "DOCKERHUB", v)
+
+				//obj, ok := v.([]map[string]string)
+				//if !ok {
+				//	logrus.WithField("obj", v).Errorf("failed to cast %s []map[string]interface{}", bucketName)
+				//} else {
+				//	Put(bucketName, bucket, "DOCKERHUB", obj)
+				//}
+			case "ssl":
+				obj, ok := v.(map[string]interface{})
+				if !ok {
+					logrus.WithField("obj", v).Errorf("failed to cast %s map[string]interface{}", bucketName)
+				} else {
+					Put(bucketName, bucket, "SSL", obj)
+				}
+			case "settings":
+				obj, ok := v.(map[string]interface{})
+				if !ok {
+					logrus.WithField("obj", v).Errorf("failed to cast %s map[string]interface{}", bucketName)
+				} else {
+					Put(bucketName, bucket, "SETTINGS", obj)
+				}
+			case "tunnel_server":
+				obj, ok := v.(map[string]interface{})
+				if !ok {
+					logrus.WithField("obj", v).Errorf("failed to cast %s map[string]interface{}", bucketName)
+				} else {
+					Put(bucketName, bucket, "INFO", obj)
+				}
+			default:
+				objlist, ok := v.([]interface{})
+				if !ok {
+					logrus.WithField("obj", v).Errorf("failed to cast %s []inferface{}", bucketName)
+				} else {
+					for _, ivalue := range objlist {
+						value, ok := ivalue.(map[string]interface{})
+						if !ok {
+							logrus.WithField("obj", value).Errorf("failed to cast %s map[string]interface{}", bucketName)
+						} else {
+							var ok bool
+							var id interface{}
+							switch bucketName {
+							case "endpoint_relations":
+								id, ok = value["EndpointID"] // TODO: need to make into an int, then do that weird stringification
+							default:
+								id, ok = value["Id"]
+							}
+							if !ok {
+								// endpoint_relations: EndpointID
+								logrus.WithField("obj", value).Errorf("No Id field:%s ", bucketName)
+								id = "error"
+							}
+							n, ok := id.(json.Number)
+							if !ok {
+								logrus.WithField("id", id).WithField("value", value).Errorf("failed to cast %s to int", bucketName)
+							} else {
+								key, err := n.Int64()
+								if err != nil {
+									logrus.WithError(err).WithField("id", id).WithField("key", key).WithField("value", value).Errorf("failed to cast %s to int", bucketName)
+								} else {
+									Put(bucketName, bucket, string(ConvertToKey(int(key))), value)
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+
+		return nil
+	})
+}
+
+// Honestly, I dunno why...
+func ConvertToKey(v int) []byte {
```

Comment:
It appears that keys need to be []byte type AFAIK from the bolt docs

Reply:
Sorry - this file has been copied as is (with 1 line modification) - from `portainer-cli` repo: https://github.com/portainer/portainer-cli/blob/master/util/database/import.go